x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/TrainedModelDefinition.java

```diff
@@ -129,21 +134,21 @@ public boolean equals(Object o) {
         if (o == null || getClass() != o.getClass()) return false;
         TrainedModelDefinition that = (TrainedModelDefinition) o;
         return Objects.equals(trainedModel, that.trainedModel) &&
-            Objects.equals(input, that.input) &&
-            Objects.equals(preProcessors, that.preProcessors);
+            Objects.equals(preProcessors, that.preProcessors) &&
+            Objects.equals(modelId, that.modelId);
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(trainedModel, input, preProcessors);
+        return Objects.hash(trainedModel, preProcessors, modelId);
     }
 
     public static class Builder {
 
         private List<PreProcessor> preProcessors;
         private TrainedModel trainedModel;
+        private String modelId;
         private boolean processorsInOrder;
```

Comment:
Unrelated to this change, but could you please explain the need for `processorsInOrder`?

Reply:
De serializing multiple named objects indicates that either they were provided in a `map` (out of order) or in an `array` (in order). We require that processors be provided in order. So, if they are provided in a map, we blow up. 

This boolean is set by the parser as it is handling the named objects. If it is not set, that means that the parser handled these as a map. If there were more than one, then we should blow up.